build/gulp/plugins/util/docgen.ts

```diff
@@ -0,0 +1,780 @@
+import * as fs from 'fs'
+import * as path from 'path'
+import * as ts from 'typescript'
+
+// currentDirectoryName is used to trim parent fileNames
```

Comment:
One small suggestion, can we document somewhere, at least with comment that these things are picked up from react-docgen-typescript? And maybe add comments where we added our changes and why they were necessary. It would be helpful, if we need in future to add something else, or fix something. Looks good otherwise, thanks for these updates, will unblock us!